src/components/user/uc/users.uc.js

```diff
@@ -0,0 +1,67 @@
+const { ObjectId } = require('mongoose').Types;
+const { GeneralError, NotFound } = require('../../../errors');
+const { userRepo, accountRepo, trashbinRepo } = require('../repo/index');
+const restrictToSameSchool = require('../../../utils/restrictToSameSchool');
+const restrictToRole = require('../../../utils/restrictToRole');
+
+const getUserData = async (id, app) => {
+	const data = {};
+
+	const user = await userRepo.getUser(id, app);
```

Comment:
In arc chapter we discuss 2 weeks ago a solution without app. Can you please look into it?

Reply:
as far as I know that solution is not ready yet, Tobi is working on it

Reply:
so for now the app should be passed along like this.

Reply:
as discussed with @Metauriel the app should be passed like this.

Reply:
Sorry but in user repo we should pass the app in user.uc and in repo to prevent to use the user model in repo? The same in account. The account repo should located at the moment in account and this repo have access to the model of the account.
In future the account is moved, ok but in this case the model and the repo is also moved. 

I strongly recommend, if we use the 3 layer structure we should prevent any app in the business and repo layer. Otherwise the code quality and the quality of the test is decrease.

If we want the populated roles from user we can use the schema action for it.